src/network/networkctl.c

```diff
@@ -673,84 +677,108 @@ static int next_lldp_neighbor(FILE *f, sd_lldp_neighbor **ret) {
+                        return r;
 
                 (void) sd_lldp_neighbor_get_system_name(n, &system_name);
                 (void) sd_lldp_neighbor_get_port_id_as_string(n, &port_id);
                 (void) sd_lldp_neighbor_get_port_description(n, &port_description);
 
-                printf("%s on port %s", strna(system_name), strna(port_id));
-
-                if (!isempty(port_description))
-                        printf(" (%s)", port_description);
+                if (asprintf(&str, "%s on port %s%s%s%s",
+                             strna(system_name), strna(port_id),
+                             isempty(port_description) ? "" : " (",
+                             port_description,
+                             isempty(port_description) ? "" : ")") < 0)
+                        return -ENOMEM;
 
-                putchar('\n');
+                r = table_add_cell(table, NULL, TABLE_STRING, str);
+                if (r < 0)
+                        return r;
 
                 c++;
         }
 
         return c;
 }
 
-static void dump_ifindexes(const char *prefix, const int *ifindexes) {
+static int dump_ifindexes(Table *table, const char *prefix, const int *ifindexes) {
         unsigned c;
+        int r;
 
         assert(prefix);
 
         if (!ifindexes || ifindexes[0] <= 0)
-                return;
+                return 0;
 
         for (c = 0; ifindexes[c] > 0; c++) {
-                char name[IF_NAMESIZE+1];
-
-                printf("%*s",
-                       (int) strlen(prefix),
-                       c == 0 ? prefix : "");
+                r = table_add_cell(table, NULL, TABLE_EMPTY, NULL);
+                if (r < 0)
+                        return r;
 
-                if (if_indextoname(ifindexes[c], name))
-                        fputs(name, stdout);
-                else
-                        printf("%i", ifindexes[c]);
+                r = table_add_cell_full(table, NULL, TABLE_STRING, c == 0 ? prefix : "", (size_t) -1, (size_t) -1, 0, 100, 0);
+                if (r < 0)
+                        return r;
 
-                fputc('\n', stdout);
+                r = table_add_cell(table, NULL, TABLE_IFINDEX, ifindexes + c);
+                if (r < 0)
+                        return r;
         }
+
+        return 0;
 }
 
-static void dump_list(const char *prefix, char **l) {
+static int dump_list(Table *table, const char *prefix, char **l) {
         char **i;
+        int r;
 
         if (strv_isempty(l))
-                return;
+                return 0;
 
         STRV_FOREACH(i, l) {
-                printf("%*s%s\n",
-                       (int) strlen(prefix),
-                       i == l ? prefix : "",
-                       *i);
+                r = table_add_cell(table, NULL, TABLE_EMPTY, NULL);
+                if (r < 0)
+                        return r;
+
+                r = table_add_cell_full(table, NULL, TABLE_STRING, i == l ? prefix : "", (size_t) -1, (size_t) -1, 0, 100, 0);
```

Comment:
Maybe use `SIZE_MAX` instead of the explicit cast? It's a bit shorter and simpler.